### PR TITLE
Wait for connection in RequestTests.ConnectionResetAbortsRequest (#978).

### PR DESCRIPTION
**Pull request by cesarbs**
The problem was that the client socket was being disposed even before the server would "see" the connection.

#978 

cc @mikeharder

### CODE REVIEW

**Comment by cesarbs** [comment]
Ping.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs [hunk 1]
```diff
@@ -211,11 +213,14 @@ public Task RemoteIPv6Address()
                 using (var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
                 {
                     socket.Connect(new IPEndPoint(IPAddress.Loopback, host.GetPort()));
+                    // Wait until connection is established
+                    Assert.True(await connectionStarted.WaitAsync(semaphoreWaitTimeout));
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs]: This is what fixes the test.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs [hunk 2]
```diff
@@ -27,6 +25,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
 {
     public class RequestTests
     {
+        const int semaphoreWaitTimeout = 2500;
```
Comment (mikeharder) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs]: Nit: `private const int`.  I would also prefix the name with `_`, but I'm not sure if this is a project convention or just my personal convention.

**Comment by mikeharder** [comment]
:shipit: after reviewing feedback

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs (hunk 2 above)
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestTests.cs]: I'd also go with underscore. We aren't the most consistent with this unfortunately, but that ssems to be the most popular for `private const`.

**Comment by halter73** [comment]
:shipit:

**Pull request merged**